pilot/pkg/features/pilot.go

```diff
@@ -655,8 +655,8 @@ var (
 	SidecarIgnorePort = env.Register("SIDECAR_IGNORE_PORT_IN_HOST_MATCH", true, "If enabled, port will not be used in vhost domain matches.").Get()
 
 	EnableEnhancedResourceScoping = env.Register("ENABLE_ENHANCED_RESOURCE_SCOPING", false,
-		"If enabled, meshConfig.discoverySelectors will also limit the configurations(like Gateway,VirtualService,DestinationRule,Ingress, etc)"+
```

Comment:
@howardjohn though we agreed on having a single selector, wondering if there should be a separate feature flag so that people get flexibility to choose. Ofcourse my usecase works with a single feature flag, just in case anyone else would need the flexibility.

Reply:
IMO no.

Either:
* Long term we want to have granular config, and should use something other than env var to expose that API
* This is only a short-term opt out, and we will remove it soon. If this is the case, there is no need for granular opt-in to new functionality.

Reply:
Makes sense